app/services/more_like_this_getter.rb

```diff
@@ -0,0 +1,87 @@
+# MoreLikeThisGetter.new(work).works
+#
+# returns an array of zero to ten
+# published works that SOLR thinks
+# are "like" the given work.
+#
+# This calls the SOLR server's more-like-this request handler
+# at http://SOME_URL/solr/SOME_CORE/mlt
+# to request a list of works that are deemed similar.
+#
+# See also:
+# work_indexer.rb for indexing details,
+# solr/config/solrconfig.xml for the request handler.
+#
+# We really do not want to slow down (or worse, break) the work show page
+# if SOLR is slow or unreachable, as it often is.
+# If anything goes wrong, you'll just get an empty array.
+#
+# MoreLikeThisGetter.new(work).works
+#
+class MoreLikeThisGetter
+  attr_reader :work
+
+  # These timeouts are short. We want to be conservative,
+  # as we don't want to trust our SOLR provider too much
+  # and this appears on a part of the website that's usually fast.
+  TIMEOUT=1
+  OPEN_TIMEOUT=1
+
+  # @param work [Work] Work
+  def initialize(work) 
+    @work = work
+  end
+
+  # Returns an array of up to 10 published works that SOLR deems similar.
+  def works
+    works_in_arbitrary_order = Work.where(
+      friendlier_id: friendlier_ids,
+      published: true)&.
+      index_by(&:friendlier_id) || {}
+    friendlier_ids.map {|id| works_in_arbitrary_order[id] }.compact
```

Comment:
This will re-fetch the works from DB every time the method is called. Is that a problem? Is the method called more than once to display the objects? Should it be memo-ized to prevent that?

Reply:
I did memoize this, yes -- good call.